addons/binding/org.openhab.binding.tankerkoenig/src/main/java/org/openhab/binding/tankerkoenig/handler/BridgeHandler.java

```diff
@@ -0,0 +1,382 @@
+import org.openhab.binding.tankerkoenig.internal.data.TankerkoenigService;
+import org.openhab.binding.tankerkoenig.internal.utility.Holiday;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link BridgeHandler} is responsible for handling the things (Tankstellen)
+ *
+ *
+ * @author Dennis Dollinger/Jürgen Baginski
+ */
+public class BridgeHandler extends BaseBridgeHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(this.getClass());
+    private String apiKey;
+    private int refreshInterval;
+    private boolean setupMode;
+    private boolean use_OpeningTime;
+
+    private ArrayList<Thing> tankstellenThingList;
+    private HashMap<String, LittleStation> tankstellenList;
+
+    private TankerkoenigListResult tankerkoenigListResult;
+
+    private ScheduledFuture<?> pollingJob;
+
+    public BridgeHandler(Bridge bridge) {
+        super(bridge);
+        tankstellenThingList = new ArrayList<Thing>();
+        tankstellenList = new HashMap<String, LittleStation>();
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        // no code needed.
+    }
+
+    @Override
+    public void initialize() {
+        Configuration config = getThing().getConfiguration();
+        setApiKey((String) config.get(TankerkoenigBindingConstants.CONFIG_API_KEY));
+        setRefreshInterval(((BigDecimal) config.get(TankerkoenigBindingConstants.CONFIG_REFRESH)).intValue());
+        setSetupMode((boolean) config.get(TankerkoenigBindingConstants.CONFIG_SETUP_MODE));
+        setUseOpeningTime((boolean) config.get(TankerkoenigBindingConstants.CONFIG_USE_OPENINGTIME));
+
+        // updateStatus(ThingStatus.ONLINE);
+        updateStatus(ThingStatus.UNKNOWN);
+
+        int pollingPeriod = this.getRefreshInterval();
+        pollingJob = scheduler.scheduleWithFixedDelay(new Runnable() {
+            @Override
+            public void run() {
+                logger.debug("Try to refresh data");
+                // Just update data if setupMode is false(off)
+                try {
+                    if (!isSetupMode()) {
+                        updateTankstellenData();
+                        updateTankstellenThings();
+                    }
+                } catch (Throwable t) {
+                    logger.error("Caught exception in ScheduledExecutorService of BridgeHandler. StackTrace: {}", t);
+                }
+            }
+        }, 30, pollingPeriod * 60, TimeUnit.SECONDS);
+        logger.debug("Refresh job scheduled to run every {} min. for '{}'", pollingPeriod, getThing().getUID());
+    }
+
+    @Override
+    public void dispose() {
+        if (pollingJob != null) {
+            pollingJob.cancel(true);
+        }
+    }
+
+    /***
+     *
+     * @param tankstelle
+     * @return
+     */
+    public boolean registerTankstelleThing(Thing tankstelle) {
+        if (tankstellenThingList.size() >= 10) {
```

Comment:
On PaperUI a 11th thing will remain OFFLINE (instead of UNKNOWN if in Setup-mode, the UNKNOWNs will go to ONLINE if this mode is deselected)
Looking bat the thing on PaperUI the user will already get displayed: 
"Status: OFFLINE - CONFIGURATION_ERROR The limitation of tankstellen things for one tankstellen config (the bridge) is limited to 10"

What else do you suggest?